lib/command.js

```diff
@@ -780,34 +797,64 @@ Expecting one of '${allowedValues.join("', '")}'`);
    */
 
   setOptionValue(key, value) {
-    return this.setOptionValueWithSource(key, value, undefined);
+    this._setPersistentOptionValueWithSource(key, value);
+    return this;
   }
 
   /**
     * Store option value and where the value came from.
     *
     * @param {string} key
     * @param {Object} value
-    * @param {string} source - expected values are default/config/env/cli/implied
+    * @param {'config'} source
     * @return {Command} `this` command for chaining
     */
 
   setOptionValueWithSource(key, value, source) {
+    this._setPersistentOptionValueWithSource(key, value, source);
+    return this;
+  }
+
+  /**
+    * Store option value and where the value came from.
+    *
+    * @param {string} key
+    * @param {Object} value
+    * @param {'default' | 'config'} [source]
+    * @api private
+    */
+
+  _setPersistentOptionValueWithSource(key, value, source) {
+    this._setNonPersistentOptionValueWithSource(key, value, source);
+    this._persistentOptionValues[key] = value;
+    this._persistentOptionValueSources[key] = source;
+  }
+
+  /**
+    * @param {string} key
+    * @param {Object} value
+    * @param {OptionValueSource} [source]
+    * @api private
+    */
+
+  _setNonPersistentOptionValueWithSource(key, value, source) {
     if (this._storeOptionsAsProperties) {
       this[key] = value;
     } else {
       this._optionValues[key] = value;
     }
     this._optionValueSources[key] = source;
-    return this;
+
+    delete this._persistentOptionValues[key];
```

Comment:
Setting a non-persistent value should not clear the underlying persistent value?

Reply:
Oh yes, you are right.

Reply:
Resolved in https://github.com/tj/commander.js/pull/1919/commits/ebf672aca9fc3187e9aa8f8161d854397f44b2ae.